### PR TITLE
[dex.py] Add len() to parsed dex structures

### PR DESCRIPTION
**Pull request by danzimm**
When using `dex.py` I found it useful to be able to calculate the size of a structure on disk. These `__len__` implementations give us that.

### CODE REVIEW

File: tools/python/dex.py [hunk 1]
```diff
@@ -1207,6 +1207,11 @@ def __init__(self, dex_class, encoded_method, is_virtual):
         self.name_in_file = None
         self.name = None
 
+    def __len__(self):
+        ci = self.get_code_item()
+        return (len(self.encoded_method) if self.encoded_method else 0) \
```
Comment (agampe) [inline tools/python/dex.py:1212]: Please run a linter.

Ours suggests:
```
        return (len(self.encoded_method) if self.encoded_method else 0) + (
            len(ci) if ci else 0
        )
```

File: tools/python/dex.py [hunk 2]
```diff
@@ -1443,6 +1448,9 @@ def __init__(self, dex, class_def):
         self.demangled = None
         self.method_mapping = None
 
+    def __len__(self):
+        return sum((len(m) for m in self.get_methods())) + sum((len(f) for f in self.get_fields()))
```
Comment (agampe) [inline tools/python/dex.py:1452]:
```
        return sum((len(m) for m in self.get_methods())) + sum(
            (len(f) for f in self.get_fields())
        )
```

File: tools/python/dex.py [hunk 3]
```diff
@@ -1613,6 +1621,9 @@ def __init__(self, dex_class, encoded_field, is_instance_field):
         self.name = None
         self.is_instance_field = is_instance_field
 
+    def __len__(self):
+        return (len(self.encoded_field) if self.encoded_field else 0)
```
Comment (agampe) [inline tools/python/dex.py:1625]: No expression parens.

**Comment by danzimm** [comment]
Thanks @agampe for pasting in the changes your guys' linter suggested. I'll run black in the future... I think that'll resolve most of these issues. Sorry about the ruckus!

**Pull request closed without merging**